waste_carriers_engine.gemspec

```diff
@@ -16,37 +16,38 @@ Gem::Specification.new do |s|
   s.description = "WasteCarriersEngine package containing shared functionality"
   s.license     = "The Open Government Licence (OGL) Version 3"
   s.metadata["rubygems_mfa_required"] = "true"
-  s.required_ruby_version = ">= 3.1.2"
+  s.required_ruby_version = ">= 3.2.2"
 
   s.files = Dir["{app,config,db,lib}/**/*", "LICENSE", "Rakefile", "README.rdoc"]
 
-  s.add_dependency "rails", ">= 6.1.4"
+  s.add_dependency "rails", "~> 7.0"
   # Use MongoDB as the database
   s.add_dependency "mongoid"
   # Implement document-level locking
-  s.add_dependency "mongoid-locker", "2.0.2"
+  # Note v2.0.1 interferes with the delegate method
+  s.add_dependency "mongoid-locker"
 
   s.add_dependency "mongo_session_store"
   # Turbolinks makes following links in your web application faster. Read more: https://github.com/rails/turbolinks
   s.add_dependency "turbolinks"
   # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
-  s.add_dependency "jbuilder", "~> 2.11"
+  s.add_dependency "jbuilder"
 
   # Use AASM to manage states and transitions
-  s.add_dependency "aasm", "~> 5.3"
```

Comment:
Same comment as for Gemfile - should some of these be pinned to their new major version number?